### PR TITLE
scanner: Stop including irrelevant file findings in `ScannerRun`

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

Fixes: #7206.

### CODE REVIEW

File: scanner/src/funTest/assets/scanner-integration-subset-pkgs-expected-ort-result.yml [hunk 1]
```diff
@@ -218,41 +218,21 @@ scanner:
       end_time: "1970-01-01T00:00:00Z"
       package_verification_code: ""
       licenses:
```
Comment (sschuberth) [inline scanner/src/funTest/assets/scanner-integration-subset-pkgs-expected-ort-result.yml:220]: Commit message typo: "which filter*s*"

File: scanner/src/main/kotlin/Scanner.kt [hunk 2]
```diff
@@ -203,16 +203,30 @@ class Scanner(
...
                 summary = scanResult.summary.copy(packageVerificationCode = "")
             )
-        }.filterTo(mutableSetOf()) { scanResult -> scanResult.provenance in relevantProvenances }
+        }.filterTo(mutableSetOf()) { scanResult ->
+            scanResult.provenance in vcsPathsForProvenances.keys
+        }.mapTo(mutableSetOf()) { scanResult ->
+            scanResult.copy(
+                summary = scanResult.summary.filterByPaths(vcsPathsForProvenances[scanResult.provenance]!!)
+            )
+        }
```
Comment (sschuberth) [inline scanner/src/main/kotlin/Scanner.kt:229]: I propose to combine the two like

        }.mapNotNullTo(mutableSetOf()) { scanResult ->
            vcsPathsForProvenances[scanResult.provenance]?.let {
                scanResult.copy(summary = scanResult.summary.filterByPaths(it))
            }
        }

which also avoids the `!!`.

File: scanner/src/main/kotlin/Scanner.kt [hunk 3]
```diff
@@ -786,3 +800,20 @@ private fun ProvenanceResolutionResult.filterByVcsPath(): ProvenanceResolutionRe
...
+    val repoPathFile = File(repositoryPath)
+    val vcsPathFile = File(vcsPath)
+
+    if (repoPathFile.startsWith(vcsPathFile)) {
+        return ""
+    } else if (vcsPathFile.startsWith(repoPathFile)) {
+        return vcsPathFile.relativeTo(repoPathFile).path
+    }
+
+    return null
```
Comment (sschuberth) [inline scanner/src/main/kotlin/Scanner.kt:818]: At a quick glance, this seem to do the same and is shorter:

    return if (repoPathFile.startsWith(vcsPathFile)) {
        ""
    } else {
        runCatching { vcsPathFile.toRelativeString(repoPathFile) }.getOrNull()
    }

**Review by sschuberth — DISMISSED** [review]

File: model/src/main/kotlin/ScanSummary.kt [hunk 4]
```diff
@@ -108,16 +108,25 @@ data class ScanSummary(
...
 
         val rootLicenseMatcher = RootLicenseMatcher(LicenseFilePatterns.getInstance())
         val applicableLicenseFiles = rootLicenseMatcher.getApplicableRootLicenseFindingsForDirectories(
             licenseFindings = licenseFindings,
-            directories = listOf(path)
+            directories = paths
         ).values.flatten().mapTo(mutableSetOf()) { it.location.path }
 
-        fun TextLocation.matchesPath() = this.path.startsWith("$path/") || this.path in applicableLicenseFiles
+        fun TextLocation.matchesPath() = paths.any { filterPath ->
```
Comment (mnonnenmacher) [inline model/src/main/kotlin/ScanSummary.kt:127]: Nit: "matchesPath**s**"

**Review by sschuberth — APPROVED** [review]

**Pull request merged**